Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -200,64 +202,102 @@ namespace AzToolsFramework
+                "Prefab - InstanceDomGenerator::GenerateContainerEntityDomFromClimbUpResult - "
+                "Called with no intance.");
 
-            AZStd::string relativePathFromRoot = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
-
-            // Creates a path from the root instance to container entity of the given insance.
-            const AZStd::string pathToContainerEntity =
-                AZStd::string::format("%s/%s", relativePathFromRoot.c_str(), PrefabDomUtils::ContainerEntityName);
-            PrefabDomPath domPathToContainerEntity(pathToContainerEntity.c_str());
+            // Get the instance's container entity DOM from the top-most ancestor that has it or from the instance itself.
+            // An ancestor DOM may not have the data if it has deleted the instance as an override
+            if (!climbUpResult.m_climbedInstances.empty())
+            {
+                const AZStd::vector<const Instance*>& climbedInstances = climbUpResult.m_climbedInstances;
+                const Instance* topInstance = climbUpResult.m_reachedInstance;
+                const Instance* const bottomInstance = climbedInstances.front();
+                for (auto instanceIter = climbedInstances.crbegin(); instanceIter != climbedInstances.crend(); ++instanceIter)
+                {
+                    // Create a path from the current top instance to the container entity of the bottom instance.
+                    AZStd::string relativePathFromTopInstance =
+                        PrefabInstanceUtils::GetRelativePathBetweenInstances(*topInstance, *bottomInstance);
+                    const AZStd::string pathToContainerEntity =
+                        AZStd::string::format("%s/%s", relativePathFromTopInstance.c_str(), PrefabDomUtils::ContainerEntityName);
+                    PrefabDomPath domPathToContainerEntity(pathToContainerEntity.c_str());
+                    const PrefabDom& topInstanceTemplateDom =
+                        m_prefabSystemComponentInterface->FindTemplateDom((*topInstance).GetTemplateId());
+                    // DOM reference can't be relied upon if the original DOM gets modified after reference creation.
+                    // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
+                    {
+                        const PrefabDomValue* containerEntityDomFromTop = domPathToContainerEntity.Get(topInstanceTemplateDom);
+                        if (containerEntityDomFromTop)
+                        {
+                            containerEntityDom.CopyFrom(*containerEntityDomFromTop, containerEntityDom.GetAllocator());
+                            break;
+                        }
+                    }
+
+                    topInstance = *instanceIter;
+                }
+            }
 
-            // Retrieves the container entity DOM of the root template DOM.
+            if (!containerEntityDom.IsObject())
```

Comment:
this can be an else or 'else if' as this can only happen if it didn't enter the other if block

Reply:
I'll add more descriptive comments, but this is not set up as an "else" in case the container entity DOM wasn't found in any of its ancestors.